### PR TITLE
Introducing Associativity GUC

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
Introduce a GUC to enable / disable associativity in ORCA. Currently, its controlled by a xform, which is not a user friendly way.
Also, set the value of the GUC to disable associativity by default.

### CODE REVIEW

File: src/test/regress/expected/qp_correlated_query_optimizer.out [hunk 1]
```diff
@@ -318,50 +318,47 @@ select A.j from A, B, C where A.j = (select C.j from C where C.j = A.j and C.i i
...
- Limit  (cost=0.00..1390613100118.43 rows=4 width=12)
-   ->  Gather Motion 3:1  (slice5; segments: 3)  (cost=0.00..1390613100118.43 rows=10 width=12)
+                                                                                                                                                                                                                            QUERY PLAN                                                                                                                                                                                                                            
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ Limit  (cost=0.00..1422599917800628.75 rows=4 width=12)
```
Comment (bhuvnesh2703) [inline src/test/regress/expected/qp_correlated_query_optimizer.out]: Even though the cost has increased, the execution time of the queries in this file decreased by 1 to 2 secs for the query with around 5k data in B and C. 
From the test query if the subquery `(select C.j from C where C.j = A.j and C.i not in (select B.i from B where C.i = B.i and B.i !=10))` represent A'. The current plan is (A' X B) X C. Older plan was (B X C) X A'.
If i modify the data and increase the rows returned by A' we get , we still get the old plan. This indicates that we may have to tune the cost model.

All the plan changes down below in the qp_correlated_query test suite for other queries are similar.

File: src/backend/gpopt/config/CConfigParamMapping.cpp [hunk 2]
```diff
@@ -493,6 +493,11 @@ CConfigParamMapping::PbsPack
 	pbs->Union(pbsJoinHeuristic);
 	pbsJoinHeuristic->Release();
 
+	if (!optimizer_enable_associativity)
```
Comment (hardikar) [inline src/backend/gpopt/config/CConfigParamMapping.cpp]: What is the behavior you want when optimizer_join_order=exhaustive & optimizer_enable_associativity=off?

Let's add a comment here:
```
// disable JoinAssociativity transform if the corresponding GUC is turned off independent of the join order algorithm chosen above
```

File: src/test/regress/expected/qp_correlated_query_optimizer.out (hunk 1 above)
Comment (hardikar) [reply src/test/regress/expected/qp_correlated_query_optimizer.out]: This increase is about 1024 times.

**Comment by bhuvnesh2703** [comment]
@hardikar. Updated the commit messages.
The plan cost increases due to the value of `optimizer_nestloop_factor=1024`, if you set the value of the GUC `optimizer_nestloop_factor=1`, the plan before or after the patch remains same.

**Pull request merged**